### PR TITLE
agent: memif support

### PR DESCRIPTION
**Pull request by sknat**
Signed-off-by: Nathan Skrzypczak <nathan.skrzypczak@gmail.com>

### CODE REVIEW

File: vpplink/types/memif.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	"github.com/projectcalico/vpp-dataplane/vpplink/binapi/vppapi/memif"
+)
+
+type MemifRole uint32
+type MemifMode uint32
+
+const (
+	MemifMaster MemifRole = MemifRole(memif.MEMIF_ROLE_API_MASTER)
+	MemifSlave  MemifRole = MemifRole(memif.MEMIF_ROLE_API_MASTER)
```
Comment (ivan4th) [inline vpplink/types/memif.go:29]: Should be ```MemifSlave  MemifRole = MemifRole(memif.MEMIF_ROLE_API_SLAVE)``` here

Comment (sknat) [reply vpplink/types/memif.go:29]: Right 😄
Thanks for the proof-read !

File: calico-vpp-agent/cni/cni_server.go [hunk 2]
```diff
@@ -87,7 +95,53 @@ func NewLocalPodSpecFromAdd(request *pb.AddRequest) (*storage.LocalPodSpec, erro
 		// for a tun it doesn't make sense
 		podSpec.ContainerIps = append(podSpec.ContainerIps, storage.LocalIP{IP: containerIp})
 	}
-
+	workload := request.GetWorkload()
+	if workload != nil {
+		for k, v := range workload.Annotations {
```
Comment (AloysAugustin) [inline calico-vpp-agent/cni/cni_server.go:100]: Let's update the annotation format as discussed offline 🙂

**Review by AloysAugustin — CHANGES_REQUESTED** [review]

File: test/yaml/Makefile [hunk 3]
```diff
@@ -1,11 +1,8 @@
 
 all: iperf-image iperf3-image wrk-image nginx-image scalepods-image scalesvc-build
 
-iperf-image:
-	docker build -t sknat/iperf -f ./iperf/Dockerfile ./iperf
-
-iperf3-image:
-	docker build -t sknat/iperf3 -f ./iperf3/Dockerfile ./iperf3
+%-image:
```
[the remark this answers is not in the document]
Comment (sknat) [reply test/yaml/Makefile:4]: Right, will rename everything `calicovpp/test-*` and redo the pushes

**Review by sknat — COMMENTED** [review]
Everythings should be addressed, updates pushed to #207 as both PRs overlap more and more

**Comment by sknat** [comment]
Closing in favor of #207

**Pull request closed without merging**